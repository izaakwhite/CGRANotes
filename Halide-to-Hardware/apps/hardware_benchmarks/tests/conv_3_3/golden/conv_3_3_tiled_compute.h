#pragma once
#include "hw_classes.h"
#include "clockwork_standard_compute_units.h"


//store is: hw_input_global_wrapper.stencil(hw_input_global_wrapper_s0_x, hw_input_global_wrapper_s0_y) = hw_input.stencil(hw_input_global_wrapper_s0_x, hw_input_global_wrapper_s0_y)
hw_uint<16> hcompute_hw_input_global_wrapper_stencil(hw_uint<16>& hw_input_stencil) {
  uint16_t _hw_input_stencil_1 = (uint16_t) hw_input_stencil.extract<0, 15>();

  return _hw_input_stencil_1;
}

//store is: conv.stencil(conv_s0_x, conv_s0_y) = (uint16)0
hw_uint<16> hcompute_conv_stencil() {
  uint16_t _284 = (uint16_t)(0);
  return _284;
}

//store is: conv.stencil(conv_s1_x, conv_s1_y) = ((hw_input_global_wrapper.stencil(conv_s1_x, conv_s1_y)*(uint16)11) + (conv.stencil(conv_s1_x, conv_s1_y) + ((hw_input_global_wrapper.stencil((conv_s1_x + 1), conv_s1_y)*(uint16)14) + ((hw_input_global_wrapper.stencil((conv_s1_x + 2), conv_s1_y)*(uint16)17) + ((hw_input_global_wrapper.stencil(conv_s1_x, (conv_s1_y + 1))*(uint16)12) + ((hw_input_global_wrapper.stencil((conv_s1_x + 1), (conv_s1_y + 1))*(uint16)255) + ((hw_input_global_wrapper.stencil((conv_s1_x + 2), (conv_s1_y + 1))*(uint16)18) + ((hw_input_global_wrapper.stencil(conv_s1_x, (conv_s1_y + 2))*(uint16)13) + ((hw_input_global_wrapper.stencil((conv_s1_x + 2), (conv_s1_y + 2))*(uint16)19) + (hw_input_global_wrapper.stencil((conv_s1_x + 1), (conv_s1_y + 2))*(uint16)16))))))))))
hw_uint<16> hcompute_conv_stencil_1(hw_uint<16>& conv_stencil, hw_uint<144>& hw_input_global_wrapper_stencil) {
  uint16_t _conv_stencil_1 = (uint16_t) conv_stencil.extract<0, 15>();

  uint16_t _hw_input_global_wrapper_stencil_1 = (uint16_t) hw_input_global_wrapper_stencil.extract<0, 15>();
  uint16_t _hw_input_global_wrapper_stencil_2 = (uint16_t) hw_input_global_wrapper_stencil.extract<16, 31>();
  uint16_t _hw_input_global_wrapper_stencil_3 = (uint16_t) hw_input_global_wrapper_stencil.extract<32, 47>();
  uint16_t _hw_input_global_wrapper_stencil_4 = (uint16_t) hw_input_global_wrapper_stencil.extract<48, 63>();
  uint16_t _hw_input_global_wrapper_stencil_5 = (uint16_t) hw_input_global_wrapper_stencil.extract<64, 79>();
  uint16_t _hw_input_global_wrapper_stencil_6 = (uint16_t) hw_input_global_wrapper_stencil.extract<80, 95>();
  uint16_t _hw_input_global_wrapper_stencil_7 = (uint16_t) hw_input_global_wrapper_stencil.extract<96, 111>();
  uint16_t _hw_input_global_wrapper_stencil_8 = (uint16_t) hw_input_global_wrapper_stencil.extract<112, 127>();
  uint16_t _hw_input_global_wrapper_stencil_9 = (uint16_t) hw_input_global_wrapper_stencil.extract<128, 143>();

  uint16_t _287 = (uint16_t)(11);
  uint16_t _288 = _hw_input_global_wrapper_stencil_1 * _287;
  uint16_t _289 = (uint16_t)(14);
  uint16_t _290 = _hw_input_global_wrapper_stencil_2 * _289;
  uint16_t _291 = (uint16_t)(17);
  uint16_t _292 = _hw_input_global_wrapper_stencil_3 * _291;
  uint16_t _293 = (uint16_t)(12);
  uint16_t _294 = _hw_input_global_wrapper_stencil_4 * _293;
  uint16_t _295 = (uint16_t)(255);
  uint16_t _296 = _hw_input_global_wrapper_stencil_5 * _295;
  uint16_t _297 = (uint16_t)(18);
  uint16_t _298 = _hw_input_global_wrapper_stencil_6 * _297;
  uint16_t _299 = (uint16_t)(13);
  uint16_t _300 = _hw_input_global_wrapper_stencil_7 * _299;
  uint16_t _301 = (uint16_t)(19);
  uint16_t _302 = _hw_input_global_wrapper_stencil_8 * _301;
  uint16_t _303 = (uint16_t)(16);
  uint16_t _304 = _hw_input_global_wrapper_stencil_9 * _303;
  uint16_t _305 = _302 + _304;
  uint16_t _306 = _300 + _305;
  uint16_t _307 = _298 + _306;
  uint16_t _308 = _296 + _307;
  uint16_t _309 = _294 + _308;
  uint16_t _310 = _292 + _309;
  uint16_t _311 = _290 + _310;
  uint16_t _312 = _conv_stencil_1 + _311;
  uint16_t _313 = _288 + _312;
  return _313;
}

//store is: hw_output.stencil(hw_output_s0_x_xi, hw_output_s0_y_yi) = conv.stencil(hw_output_s0_x_xi, hw_output_s0_y_yi)
hw_uint<16> hcompute_hw_output_stencil(hw_uint<16>& conv_stencil) {
  uint16_t _conv_stencil_2 = (uint16_t) conv_stencil.extract<0, 15>();

  return _conv_stencil_2;
}

