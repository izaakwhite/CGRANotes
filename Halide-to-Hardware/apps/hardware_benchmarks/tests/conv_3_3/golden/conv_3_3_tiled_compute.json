{
"namespaces":{
  "global":{
    "modules":{
      "hcompute_conv_stencil":{
        "type":["Record",[
          ["out_conv_stencil",["Array",16,"Bit"]]
        ]],
        "instances":{
          "const_p0__285":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0000"]}
          }
        },
        "connections":[
          ["self.out_conv_stencil","const_p0__285.out"]
        ]
      },
      "hcompute_conv_stencil_1":{
        "type":["Record",[
          ["out_conv_stencil",["Array",16,"Bit"]],
          ["in0_conv_stencil",["Array",1,["Array",16,"BitIn"]]],
          ["in1_hw_input_global_wrapper_stencil",["Array",9,["Array",16,"BitIn"]]]
        ]],
        "instances":{
          "add_315_339_340":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_317_337_338":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_319_336_337":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_321_335_336":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_323_334_335":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_325_333_334":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_327_332_333":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_329_331_332":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_conv_stencil_1_338_339":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "const_p11__314":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h000b"]}
          },
          "const_p12__320":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h000c"]}
          },
          "const_p13__326":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h000d"]}
          },
          "const_p14__316":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h000e"]}
          },
          "const_p16__330":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0010"]}
          },
          "const_p17__318":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0011"]}
          },
          "const_p18__324":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0012"]}
          },
          "const_p19__328":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0013"]}
          },
          "const_p255__322":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h00ff"]}
          },
          "mul_hw_input_global_wrapper_stencil_1_314_315":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_2_316_317":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_3_318_319":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_4_320_321":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_5_322_323":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_6_324_325":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_7_326_327":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_8_328_329":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_9_330_331":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          }
        },
        "connections":[
          ["mul_hw_input_global_wrapper_stencil_1_314_315.out","add_315_339_340.in0"],
          ["add_conv_stencil_1_338_339.out","add_315_339_340.in1"],
          ["self.out_conv_stencil","add_315_339_340.out"],
          ["mul_hw_input_global_wrapper_stencil_2_316_317.out","add_317_337_338.in0"],
          ["add_319_336_337.out","add_317_337_338.in1"],
          ["add_conv_stencil_1_338_339.in1","add_317_337_338.out"],
          ["mul_hw_input_global_wrapper_stencil_3_318_319.out","add_319_336_337.in0"],
          ["add_321_335_336.out","add_319_336_337.in1"],
          ["mul_hw_input_global_wrapper_stencil_4_320_321.out","add_321_335_336.in0"],
          ["add_323_334_335.out","add_321_335_336.in1"],
          ["mul_hw_input_global_wrapper_stencil_5_322_323.out","add_323_334_335.in0"],
          ["add_325_333_334.out","add_323_334_335.in1"],
          ["mul_hw_input_global_wrapper_stencil_6_324_325.out","add_325_333_334.in0"],
          ["add_327_332_333.out","add_325_333_334.in1"],
          ["mul_hw_input_global_wrapper_stencil_7_326_327.out","add_327_332_333.in0"],
          ["add_329_331_332.out","add_327_332_333.in1"],
          ["mul_hw_input_global_wrapper_stencil_8_328_329.out","add_329_331_332.in0"],
          ["mul_hw_input_global_wrapper_stencil_9_330_331.out","add_329_331_332.in1"],
          ["self.in0_conv_stencil.0","add_conv_stencil_1_338_339.in0"],
          ["mul_hw_input_global_wrapper_stencil_1_314_315.in1","const_p11__314.out"],
          ["mul_hw_input_global_wrapper_stencil_4_320_321.in1","const_p12__320.out"],
          ["mul_hw_input_global_wrapper_stencil_7_326_327.in1","const_p13__326.out"],
          ["mul_hw_input_global_wrapper_stencil_2_316_317.in1","const_p14__316.out"],
          ["mul_hw_input_global_wrapper_stencil_9_330_331.in1","const_p16__330.out"],
          ["mul_hw_input_global_wrapper_stencil_3_318_319.in1","const_p17__318.out"],
          ["mul_hw_input_global_wrapper_stencil_6_324_325.in1","const_p18__324.out"],
          ["mul_hw_input_global_wrapper_stencil_8_328_329.in1","const_p19__328.out"],
          ["mul_hw_input_global_wrapper_stencil_5_322_323.in1","const_p255__322.out"],
          ["self.in1_hw_input_global_wrapper_stencil.0","mul_hw_input_global_wrapper_stencil_1_314_315.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.1","mul_hw_input_global_wrapper_stencil_2_316_317.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.2","mul_hw_input_global_wrapper_stencil_3_318_319.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.3","mul_hw_input_global_wrapper_stencil_4_320_321.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.4","mul_hw_input_global_wrapper_stencil_5_322_323.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.5","mul_hw_input_global_wrapper_stencil_6_324_325.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.6","mul_hw_input_global_wrapper_stencil_7_326_327.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.7","mul_hw_input_global_wrapper_stencil_8_328_329.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.8","mul_hw_input_global_wrapper_stencil_9_330_331.in0"]
        ]
      },
      "hcompute_hw_input_global_wrapper_stencil":{
        "type":["Record",[
          ["out_hw_input_global_wrapper_stencil",["Array",16,"Bit"]],
          ["in0_hw_input_stencil",["Array",1,["Array",16,"BitIn"]]]
        ]],
        "connections":[
          ["self.out_hw_input_global_wrapper_stencil","self.in0_hw_input_stencil.0"]
        ]
      },
      "hcompute_hw_output_stencil":{
        "type":["Record",[
          ["out_hw_output_stencil",["Array",16,"Bit"]],
          ["in0_conv_stencil",["Array",1,["Array",16,"BitIn"]]]
        ]],
        "connections":[
          ["self.out_hw_output_stencil","self.in0_conv_stencil.0"]
        ]
      }
    }
  }
}
}
